app/models/core/IncomeSourceIdHash.scala

```
@@ -38,8 +38,8 @@ object IncomeSourceIdHash {
     new IncomeSourceIdHash(s"${hashB}${hashA}")
   }
 
-  def mkFromQueryString(hashCodeAsString: String): Either[Throwable, IncomeSourceIdHash] = Try {
+  def mkFromQueryString(hashCodeAsString: String): Option[IncomeSourceIdHash] = Try {
```

I think we can keep original method signature as it was, and convert Either to Option where we need this, as we rely suppressing error details here

---

If we convert to option we suppress the error details anyway

---

yes, but this would decided by the caller and can still go back to preserver actual error in the future